### PR TITLE
[MER-743] Add ability to delete a section

### PR DESCRIPTION
**Pull request by gastonabella**
[MER-743](https://eliterate.atlassian.net/browse/MER-743)

Add button in the Sections Overview page to delete (if there's no student associated data) a section or archive it.

### CODE REVIEW

File: lib/oli/delivery/sections.ex [hunk 1]
```diff
@@ -273,21 +273,44 @@ defmodule Oli.Delivery.Sections do
...
+  Returns a listing of all student enrollments for a given section.
+
+  """
+  def list_student_enrollments(section_slug) do
+    learner_role_id = ContextRoles.get_role(:context_learner).id
+
+    query =
+      from(
+        e in Enrollment,
+        join: s in Section,
```
Comment (jrissler) [inline lib/oli/delivery/sections.ex]: What do you think about `join: section in assoc(e, :section)` (saves 1 line / easier to scan)

File: lib/oli/delivery/sections.ex [hunk 2]
```diff
@@ -621,6 +644,30 @@ defmodule Oli.Delivery.Sections do
...
 
+  @doc """
+  Archives a section by marking the record as archived.
+  ## Examples
+      iex> archive_section(section)
+      {:ok, %Section{}}
+      iex> archive_section(section)
+      {:error, %Ecto.Changeset{}}
+  """
+  def archive_section(%Section{} = section) do
```
Comment (jrissler) [inline lib/oli/delivery/sections.ex]: I feel like calling update function vs having a specific function just for this that calls the update function anyways - feels super redundant / not needed.

File: priv/repo/migrations/20220216163315_add_on_delete_to_section_associations.exs [hunk 3]
```diff
@@ -0,0 +1,105 @@
+defmodule Oli.Repo.Migrations.AddOnDeleteToSectionAssociations do
+  use Ecto.Migration
+
+  def change do
```
Comment (jrissler) [inline priv/repo/migrations/20220216163315_add_on_delete_to_section_associations.exs:4]: Love all of these - can't believe this wasn't done originally lol.

**Review by jrissler — CHANGES_REQUESTED** [review]

File: lib/oli/delivery/sections.ex (hunk 1 above)
Comment (gastonabella) [reply lib/oli/delivery/sections.ex]: Looks better :)

**Review by jrissler — APPROVED** [review]

File: lib/oli_web/live/sections/overview_view.ex [hunk 4]
```diff
@@ -144,4 +150,67 @@ defmodule OliWeb.Sections.OverviewView do
...
+          This section has student data and will be archived rather than deleted.
+          Are you sure you want to archive it? You will no longer have access to the data.
+        """
+      else
+        """
+          This action cannot be undone. Are you sure you want to delete this section?
+        """
+      end
+
+    action = if section_has_student_enrollments, do: "Archive", else: "Delete"
```
Comment (mcass19) [inline lib/oli_web/live/sections/overview_view.ex]: NIT: We can wrap this two if of the same condition into a tuple or something. Like `{message, action} = if section_has_student_enrollments do ...`

**Review by mcass19 — APPROVED** [review]
🚀 🚀 🚀

**Pull request closed without merging**